src/Resources/ResourceManager/SdkClient/ResourceManagerSdkClient.cs

```diff
@@ -433,7 +433,14 @@ private Deployment CreateBasicDeployment(PSDeploymentCmdletParameters parameters
             }
             else
             {
-                deployment.Properties.Template = JObject.Parse(FileUtilities.DataStore.ReadFileAsText(parameters.TemplateFile));
+                if (!string.IsNullOrEmpty(parameters.TemplateFile))
+                {
+                    deployment.Properties.Template = JObject.Parse(FileUtilities.DataStore.ReadFileAsText(parameters.TemplateFile));
+                }
+                else
+                {
+                    deployment.Properties.Template = JObject.Parse(JsonConvert.SerializeObject(parameters.TemplateObject));
```

Comment:
Do we know that this is filled in at this point?  Do our parameter sets require this to be non-empty if we reach this point?

Reply:
@markcowl yes, `-TemplateFile`, `-TemplateUri` and `-TemplateObject` are all mutually exclusive parameters, and `parameters.TemplateFile` only has a value if the user provides a value to the first two parameters, and `parameters.TemplateObject` only has a value if the user provided a value to `-TemplateObject`, so neither property should be `null` when the user jumps into the respective block.

Reply:
@markcowl just added the `[ValidateNotNull]` attribute to the new `-TemplateObject` parameter